src/sniffer.c

```diff
@@ -3498,10 +3504,44 @@ int ssl_DecodePacket(const byte* packet, int length, byte** data, char* error)
     ret = ProcessMessage(sslFrame, session, sslBytes, data, end, error);
     if (RemoveFatalSession(&ipInfo, &tcpInfo, session, error)) return -1;
     CheckFinCapture(&ipInfo, &tcpInfo, session);
+
+    /* Pass back Session Info after we have processed the Server Hello. */
+    if ((NULL != sslInfo) && (0 != session->sslServer->options.cipherSuite)) {
+        const char* pCipher;
+
+        sslInfo->isValid = 1;
+        sslInfo->protocolVersionMajor = session->sslServer->version.major;
+        sslInfo->protocolVersionMinor = session->sslServer->version.minor;
+        sslInfo->serverCipherSuite0 = session->sslServer->options.cipherSuite0;
+        sslInfo->serverCipherSuite = session->sslServer->options.cipherSuite;
+
+        pCipher = wolfSSL_get_cipher(session->sslServer);
+        if (NULL != pCipher)
+            XMEMCPY(sslInfo->serverCipherSuiteName, pCipher,
```

Comment:
This looks like it could overread, not all names are the max size.

Reply:
Changing to a strncpy from memcpy. Capping it at the length of the destination, and forcing a null terminator at the end of the suite name.